app/components/ActivityAdministration.jsx

```diff
@@ -0,0 +1,408 @@
+    var days = value.match(new RegExp ('([0-9]+d){1}', 'g'))
+    var other = value.match(new RegExp ('[^0-9mhd\u00a0\u0020]+', 'g'))
+    return (
+      (!!days || !!hours || !!min) && !other
+    )
+  }
+
+})
+Formsy.addValidationRule('isMoreOrGreaterIntThanZero', function (values, value) {
+  return (value % 1 === 0 && value >= 0)
+})
+
+var AddedVolonteer = React.createClass({
+  onClick: function () {
+    this.props.onRemoveButtonClick(this.props.volunteer)
+  },
+  name: function() {
+    var v = this.props.volunteer
+    return v.display_name || (v.first_name +' '+ v.last_name)
+  },
+  render: function () {
+    return (
+        <div className="addedVolonteer" ><a href={'/wolontariusz/'+this.props.volunteer.user_id}>{this.name()}</a> <input type="button" className="addedVolonteerRemoveButton" onClick={this.onClick} value="Usuń"/></div>
+    )
+  }
+})
+
+var ActivityAdministration = React.createClass({
+
+  getInitialState: function () {
+    var state = this.props.context.getStore(ActivityStore).getState()
+    // Tworzy kopię tablicy volontariuszy po to żeby później stwierdzić jakie
+    // zaszły wn niej zmiany.
+    state._volunteers = Object.assign({}, state).volunteers
+    return state
+  },
+
+  _changeListener: function() {
+    this.setState(this.props.context.getStore(ActivityStore).getState())
+  },
+
+  componentDidMount: function() {
+    this.props.context.getStore(ActivityStore)
+      .addChangeListener(this._changeListener)
+  },
+
+  componentWillUnmount: function() {
+    this.props.context.getStore(ActivityStore)
+      .removeChangeListener(this._changeListener)
+  },
+
+  user: function() {
+    return this.props.context.getUser()
+  },
+
+  handleStartEventTimestampChange: function (m) {
+    //http://stackoverflow.com/questions/18022534/moment-js-and-unix-epoch-conversion
+    this.setState(update(this.state, {
+      activity: {startEventTimestamp: {$set: m.toDate().getTime()}}
+    }))
+  },
+
+  handleChange: function (evt) {
+    var activity = {}
+    var target = evt.target != null
+      ? evt.target
+      : evt.currentTarget
+    var value = target.type === 'checkbox'
+      ? target.checked
+      : target.value
+    activity[target.name] = {$set: value}
+    this.setState(update(this.state, {
+      activity: activity
+    }))
+  },
+
+  addActiveVolonteer: function (volunteer) {
+   var joint = {
+     user_id: volunteer.user_id,
+     activity_id: this.state.activity.id
+    }
+    
+    if (this.state.volunteers.length == this.state.activity.maxVolunteers) {
+      this.setState(update(this.state, {
+        invalidSnackBar: {$set: 'Osiągnięto limit wolontariuszy'}
+      }))
+      return
+    }
+    
+    for (var i=0; i<this.state.volunteers.length; i++) {
+      if (this.state.volunteers[i].user_id == volunteer.user_id) {
+        this.setState(update(this.state, {
+          invalidSnackBar: {$set: 'Wolontariusz jest już dodany'}
+        }))
+      return
+      }
+    }
```

Comment:
Ok, może być tak jak proponujesz z tym limitem i sugestiami. Spróbuję to jeszcze dzisiaj zrobić